drivers/flash/flash_stm32h7x.c

```diff
@@ -368,6 +368,25 @@ int flash_stm32_write_range(const struct device *dev, unsigned int offset,
 	const uint8_t ndwords = FLASH_NB_32BITWORD_IN_FLASHWORD / 2;
 	const uint8_t nbytes = FLASH_NB_32BITWORD_IN_FLASHWORD * 4;
 	uint8_t unaligned_datas[nbytes];
+	uint8_t unaligned_bytes = (uintptr_t)data % nbytes;
+
+	/* First check if *data is nbytes aligned */
+	if (unaligned_bytes != 0) {
+		/* *data is not nbytes byte aligned */
+		memset(unaligned_datas, 0xff, nbytes);
```

Comment:
Not fully correct:
- I might erase existing written bytes here.
- I'm adding and offset to the expected writing address